File: clojure/raindrops/_iterations/1/src/raindrops.clj
(ns raindrops)

(defn- if-factor [num factor value]
  (if (zero? (mod num factor))
    value
    ""))

(defn- pling [num]
  (if-factor num 3 "Pling"))

(defn- plang [num]
  (if-factor num 5 "Plang"))

(defn- plong [num]
  (if-factor num 7 "Plong"))

(defn convert [num]
  (let [result (str (pling num) (plang num) (plong num))]
    (if (empty? result)
      (str num)
      result)))

File: ruby/raindrops/_iterations/1/raindrops.rb
module Raindrops
  SOUNDS = [[3, "Pling"], [5, "Plang"], [7, "Plong"]]

  def self.convert(n)
    sounds = SOUNDS.reduce("") { |s, sound| n % sound[0] == 0 ? s + sound[1] : s }
    sounds.empty? ? n.to_s : sounds
  end
end
